src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,52 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
+
+const dependenciesGlobal = {};
 
 async function getDependencies() {
-  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+  if (Object.keys(dependenciesGlobal).length === 0) {
+    const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+    // The order of the alternatives is important!
```

Comment:
Could you explain what `alternatives`  are you talking about?

Reply:
I meant alternations of the regex: `\r\n` or `\r` or `\n`.

Reply:
Then `\r\n?` should be sufficient right

Reply:
Not really, because for Linux, it is only `\n`, but here we are accepting `\r\n` and `\r`.

Reply:
I think we should use `\r?\n` instead. From this [stackoverflow answer](https://stackoverflow.com/a/1552782), the line-ending `\r` only occurs on really old Macintosh systems.

Reply:
I think we should include it, for example  in the Typescript repo
https://github.com/microsoft/TypeScript/blob/e64f04bd8c87ece7f800d10d8eb90b7e592907ad/src/harness/util.ts#L35